src/lib/operators/print.cpp

```diff
@@ -0,0 +1,110 @@
+#include "print.hpp"
+
+#include <iomanip>
+
+#include "operators/table_wrapper.hpp"
+#include "storage/abstract_segment.hpp"
+#include "storage/table.hpp"
+#include "type_cast.hpp"
+
+namespace {
+
+using namespace opossum;  // NOLINT(build/namespaces)
+
+std::string print_column_type(const std::shared_ptr<const Table>& table, const ColumnID column_id) {
+  auto stream = std::stringstream{};
+  stream << table->column_type(column_id);
+  if (table->column_nullable(column_id)) {
+    stream << "_null";
+  }
+
+  return stream.str();
+}
+
+}  // namespace
+
+namespace opossum {
+
+Print::Print(const std::shared_ptr<const AbstractOperator> in, std::ostream& out) : AbstractOperator(in), _out(out) {}
+
+void Print::print(std::shared_ptr<const Table>& table, std::ostream& out) {
+  auto table_wrapper = std::make_shared<TableWrapper>(table);
+  table_wrapper->execute();
+  Print(table_wrapper, out).execute();
+}
+
+std::shared_ptr<const Table> Print::_on_execute() {
+  auto widths = _column_string_widths(8, 20, _left_input_table());
+
+  // Print column headers.
+  _out << "=== Columns" << std::endl;
+  const auto left_column_count = _left_input_table()->column_count();
```

Comment:
I personally find always writing "left_column_count" or "left_chunk_count" a bit confusing, since it leads me to thinks that there also needs to be a right_column_cont etc. Of course when taking into consideration the AbstractOperator this is technically right, however I guess this especially would confuse somebody, who is not as involved in the Codebase